File: swift/problem720.swift
class Solution {
    func longestWord(_ words: [String]) -> String {
        let words = words.sorted{$0.count < $1.count}
        var longest_length = 1
        var records = Set<String>()
        var ans = "{" //larger than any small case string
        
        for var word in words {
            if (word.count == 1) {
                records.insert(word)
            } else {
                let tmp = word
                if (records.contains(word)) {
                    records.insert(tmp)
                    longest_length = max(longest_length,tmp.count)
                }
            }
        }
        
        for word in records where (word.count == longest_length) {
            ans = min(word,ans)
        }
        
        return ans
    }
}

File: python/problem720.py
class Solution:
    def longestWord(self, words):
        """
        :type words: List[str]
        :rtype: str
        """
        se=set(words)
        li=[]
        maxl=0
        
        for i in se:
            if len(i)<maxl:
                continue
            
            for k in range(1,len(i)):
                if i[0:k] not in se:
                    break
                
                if k==len(i)-1:
                    if maxl>0 and len(i)>maxl:
                        li.clear()
                    li.append(i)
                    maxl=len(i)
        if li:
            return min(li)
        else:
            return min(se)